qulice-checkstyle/src/main/java/com/qulice/checkstyle/JavadocEmptyLineCheck.java

```diff
@@ -0,0 +1,136 @@
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+
+/**
+ * Check for empty line at the start and at the end of javadoc.
+ * @author Denys Skalenko (d.skalenko@gmail.com)
+ * @version $Id$
+ * @since 0.17
+ */
+public final class JavadocEmptyLineCheck extends Check {
+
+    @Override
+    public int[] getDefaultTokens() {
+        return new int[] {
+            TokenTypes.CLASS_DEF,
+            TokenTypes.INTERFACE_DEF,
+            TokenTypes.VARIABLE_DEF,
+            TokenTypes.CTOR_DEF,
+            TokenTypes.METHOD_DEF,
+        };
+    }
+
+    @Override
+    public void visitToken(final DetailAST ast) {
+        final String[] lines = this.getLines();
+        final int start = ast.getLineNo();
+        final int cstart =
+            JavadocEmptyLineCheck.findCommentStart(lines, start) + 1;
+        if (JavadocEmptyLineCheck.isNodeHasJavaDoc(ast, cstart)) {
+            if (JavadocEmptyLineCheck.isJavadocLineEmpty(lines[cstart])) {
+                this.log(cstart + 1, "Empty javadoc line at the beginning");
+            }
+            final int cend =
+                JavadocEmptyLineCheck.findCommentEnd(lines, start) - 1;
+            if (JavadocEmptyLineCheck.isJavadocLineEmpty(lines[cend])) {
+                this.log(cend + 1, "Empty javadoc line at the end");
+            }
+        }
+    }
+
+    /**
+     * Check if javadoc Line Empty.
+     * @param line Java doc line
+     * @return Is Javadoc Line Empty
+     */
+    private static boolean isJavadocLineEmpty(final String line) {
+        return null != line && "*".equals(line.trim());
+    }
+
+    /**
+     * Check if node has java doc.
+     * @param node Node to be checked for Java docs.
+     * @param start Line number where comment starts.
+     * @return Is node has java doc
+     */
+    private static boolean isNodeHasJavaDoc(final DetailAST node,
+        final int start) {
+        int cprevious = 0;
+        final DetailAST previous = node.getPreviousSibling();
+        if (null != previous) {
+            cprevious = previous.getLineNo();
+        }
+        return start > cprevious;
+    }
+
+    /**
+     * Find javadoc starting comment.
+     * @param lines List of lines to check.
+     * @param start Start searching from this line number.
+     * @return Line number with found starting comment or -1 otherwise.
+     */
+    private static int findCommentStart(final String[] lines, final int start) {
+        return JavadocEmptyLineCheck.findTrimmedTextUp(lines, start, "/**");
+    }
+
+    /**
+     * Find javadoc ending comment.
+     * @param lines List of lines to check.
+     * @param start Start searching from this line number.
+     * @return Line number with found ending comment, or -1 if it wasn't found.
+     */
+    private static int findCommentEnd(final String[] lines, final int start) {
+        return JavadocEmptyLineCheck.findTrimmedTextUp(lines, start, "*/");
```

Comment:
@krzyk Could you please rephrase the question, because I can't understand what is wrong here, for me both cases valid and JavadocEmptyLineCheck should not prohibit